docs/ml-features.md

```diff
@@ -183,6 +183,75 @@ for words_label in wordsDataFrame.select("words", "label").take(3):
 </div>
 </div>
 
+## OneHotEncoder
+
+One-hot encoding is a way of formatting categorical features as input into machine learning algorithms. It maps a column of label indices to a column of binary vectors, with at most a single one-value. The [OneHotEncoder](api/scala/index.html#org.apache.spark.ml.feature.OneHotEncoder) class provides this functionality. By default, the resulting binary vector has a component for each category, so with 5 categories, an input value of 2.0 would map to an output vector of (0.0, 0.0, 1.0, 0.0, 0.0). If the includeFirst is set to false, the first category is omitted, so the output vector for the previous example would be (0.0, 1.0, 0.0, 0.0) and an input value of 0.0 would map to a vector of all zeros. Including the first category makes the vector columns linearly dependent because they sum up to one.
+
+<div class="codetabs">
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+import org.apache.spark.ml.feature.{OneHotEncoder, StringIndexer}
+
+val df = sqlContext.createDataFrame(Seq(
+  (0, "a"),
+  (1, "b"),
+  (2, "c"),
+  (3, "a"),
+  (4, "a"),
+  (5, "c")
+)).toDF("id", "category")
+
+val indexer = new StringIndexer().setInputCol("category").
```

Comment:
Scala style (here and elsewhere in examples): either 1 setter per line:
```
val indexer = new StringIndexer()
  .setInputCol("category")
  .setOutputCol("categoryIndex")
  .fit(df)
```
or compact (if it fits):
```
val indexer = new StringIndexer().setInputCol("category").setOutputCol("categoryIndex").fit(df)
```

Also, does it help to include StringIndexer?  I like keeping these examples as simple as possible, so perhaps you could start with indices from the start.  We can include integration examples in the examples/ directory code.